### ISSUE TITLE
drop_failed_sims() checks only x, so a non-finite theta reaches chol()

### ISSUE THREAD

**Issue by pedroliman** [pedroliman/neuralsbi#38]
`drop_failed_sims()` (`R/simulator.R:234-251`) decides which rows survive from `x` alone:

```r
ok <- rowSums(!is.finite(x)) == 0L
```

`theta` is then subset by that mask and returned unfiltered for its own contents (`:249`). The docstring says "`theta` and `x` go together so the training set stays paired", which is true of the pairing but not of the finiteness check.

Pre-computed simulations are the path that matters, since `npe(prior, theta = , x = )` runs the same check (`R/npe.R:168`) and it is the user who supplied `theta`:

```r
th <- sample_prior(prior, 200); th[3, 1] <- NA
npe(prior, theta = th, x = xx, density_estimator = "linear_gaussian")
#> Error: the leading minor of order 1 is not positive
```

That message comes from `chol()` at `R/density_estimator.R:49`. On the torch path it becomes "Training failed: no restart produced a finite validation loss" (`R/train.R:157`), which blames training for a bad input.

## Fix

Test finiteness on both matrices. The warning already reports a count and rate, so it can say which side failed. Keep the "all failed" error. Regression test on the pre-computed path with one `NA` in `theta`, asserting the row is dropped and the warning names the count.
